speculos.py

```diff
@@ -63,19 +65,23 @@ def run_qemu(s1, s2, app_path, libraries=[], seed=DEFAULT_SEED, debug=False, tra
     if trace_syscalls:
         args += [ '-t' ]
 
-    if rampage is not None and pagesize is not None:
-        args += [ '-r' , rampage ] + [ '-s' , pagesize ]
-
     if model is not None:
         args += ['-m', model]
 
     args += [ '-k', str(sdk_version) ]
 
     for lib in [ f'main:{app_path}' ] + libraries:
         name, lib_path = lib.split(':')
-        load_offset, load_size, stack, stack_size = get_elf_infos(lib_path)
+        load_offset, load_size, stack, stack_size, ram_addr, ram_size = get_elf_infos(lib_path)
         args.append(f'{name}:{lib_path}:{load_offset}:{load_size}:{stack}:{stack_size}')
 
+    if rampage is not None and pagesize is not None:
+        print('[+] using user-provided additional RAM page')
+        args += [ '-r' , rampage ] + [ '-s' , pagesize ]
+    else:
```

Comment:
if there's no bssrf, i.e. for nanos application we should not add these parameters

Reply:
OK

Reply:
Fixed. If no such page was declared when the app was linked (i.e. any Nano or Blue app built against an SDK the linker script of which did not declare such page e.g. SDK 1.5), then auto-probing returns 0 and no additional page is used.